6MI0800193/Week1/Task3.cpp

```diff
@@ -0,0 +1,103 @@
+#include <iostream>
+using namespace std;
+enum class Year {
+    first=1,
+    second=2,
+    third=3,
+    fourth=4
+};
+
+struct Student
+{
+    char name[16];
+    unsigned int fn;
+    Year school_year;
+    double grade;
+};
+
+void print_student(const Student& student)
+{
+    cout << student.name << " ";
+    cout << student.fn << " ";
+    switch (student.school_year)
+    {
+    case Year::first:
+        cout << "first" << " ";
+        break;
+    case Year::second:
+        cout << "second" << " ";
+        break;
+    case Year::third:
+        cout << "third" << " ";
+        break;
+    case Year::fourth:
+        cout << "fourth" << " ";
+        break;
+    }
+
+    cout << student.grade << endl;
+}
+
+bool init_student(Student& toInit, char name[16], unsigned int facNumber, Year currentCourse, double grade)
+{
+    if (facNumber < 10000 || facNumber > 99999)
+        return false;
+
+    if ((int)currentCourse < 1 || (int)currentCourse > 4)
+        return false;
+
+    if (grade < 3 || grade > 6)
+        return false;
+
+    toInit.fn = facNumber;
+
+    for (size_t i = 0; i < 16; i++)
+    {
+        toInit.name[i] = name[i];
+    }
+
+    toInit.school_year = currentCourse;
+    toInit.grade = grade;
+}
+
+void print_scholars(const Student* students, size_t length, double average)
+{
+    for (size_t i = 0; i < length; i++)
+    {
+        if (students[i].grade > average)
+            print_student(students[i]);
+    }
+}
+
+void print_scholars_sorted(Student* students, size_t length, double average)

```

Comment:
Wouldn't this look better as a separate function named `sort` ? This way you'll have the current one re-written as 
```
void print_scholars_sorted(Student* students, size_t length, double average)
{
    sort(students, length);
     print_scholars(students, length, average);
}
```

and the `sort` you may re-use eventually :))